### PR TITLE
More video solutions for silver questions p738

### PR DESCRIPTION
**Pull request by cskitty**

### CODE REVIEW

File: solutions/usaco-738.mdx [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+id: usaco-738
+source: USACO Silver 2017 US Open
+title: Paired Up
+author: Melody Yu
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_pairup_silver_open17.html)
+
+## Video Solution
+
```
Comment (nwatx) [inline solutions/usaco-738.mdx:11]: would be beneficial to have the code here, but good video.

**Review by nwatx — CHANGES_REQUESTED** [review]

File: solutions/usaco-738.mdx (hunk 1 above)
Comment (nwatx) [reply solutions/usaco-738.mdx:11]: or at least linked from the video

**Review by nwatx — CHANGES_REQUESTED** [review]

File: solutions/usaco-738.mdx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+		cows[i] = {a, b};
+	}
+
+	int currL = 0;
+	int countL = 0;
+	int currR = N - 1;
+	int countR = 0;
+	ll ans = 0;
+
+	sort(cows.begin(), cows.end(), [](pair<ll, ll> a, pair<ll, ll> b) {return a.s < b.s;});
```
[the remark this answers is not in the document]
Comment (frodakcin) [reply solutions/usaco-738.mdx:120]: I think an exception can be made here -- It's not exactly practical to shorten this.

[the remark this answers is not in the document]
Comment (maggie-j-liu) [reply solutions/usaco-738.mdx:120]: try putting the function body on a new line like
```cpp
	sort(cows.begin(), cows.end(), [](pair<ll, ll> a, pair<ll, ll> b) {
		return a.s < b.s;
	});
```

[the remark this answers is not in the document]
Comment (nwatx) [reply solutions/usaco-738.mdx:120]: alternatively you could use `auto`

File: solutions/usaco-738.mdx [hunk 3]
```diff
@@ -0,0 +1,142 @@
+---
+id: usaco-738
+source: USACO Silver 2017 US Open
+title: Paired Up
+author: Sofia Yang
```
Comment (maggie-j-liu) [inline solutions/usaco-738.mdx:5]: add your name to authors + remove blank line

File: solutions/usaco-738.mdx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+(Because it requires sorting the input, which is $\mathcal{O}(N\log N)$).
+
+
+<LanguageSection>
+
+<JavaSection>
+
+
+``` java
```
Comment (maggie-j-liu) [inline solutions/usaco-738.mdx:26]:
```suggestion
```java
```

File: solutions/usaco-738.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (frodakcin) [reply solutions/usaco-738.mdx:120]: mm sure, though personally I think the current way looks the nicest

File: solutions/usaco-738.mdx [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+</JavaSection>
+
+<CPPSection>
+
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+#define ll long long
```
Comment (maggie-j-liu) [inline solutions/usaco-738.mdx:94]: would prefer 
```suggestion
using ll = long long;
```

File: solutions/usaco-738.mdx [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+#define f first
+#define s second
+
+ll N;
```
Comment (nwatx) [inline solutions/usaco-738.mdx:98]:
```suggestion
int N;
```

**Review by nwatx — CHANGES_REQUESTED** [review]

File: solutions/usaco-738.mdx [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+[Official Analysis](http://www.usaco.org/current/data/sol_pairup_silver_open17.html)
+
+## Video Solution
+
+Note: The video solution might not be the same as other solutions. Code in C++.
+<Youtube id="yIoUWlzH_7w" />
+
+**Time Complexity:** $\mathcal{O}(N\log N)$
+
+(Because it requires sorting the input, which is $\mathcal{O}(N\log N)$).
```
Comment (frodakcin) [inline solutions/usaco-738.mdx:18]:
```suggestion
**Time Complexity:** $\mathcal{O}(N)$ after sorting
```

**Review by frodakcin — CHANGES_REQUESTED** [review]

File: solutions/usaco-738.mdx [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+		}
+	}
+
+	cout << ans;
+}
+```
+
+</CPPSection>
+
+
```
Comment (maggie-j-liu) [inline solutions/usaco-738.mdx:141]:
```suggestion
```

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]
unrelated, but: why's the diff so weird? it looks like you added the whole file but the java code was there before???

**Comment by bqi343** [comment]
@frodakcin @maggie-j-liu

**Review by maggie-j-liu — APPROVED** [review]
i think you might need to fix the merge conflicts though

**Review by nwatx — APPROVED** [review]

**Comment by maggie-j-liu** [comment]
@cskitty please fix the merge conflicts

**Review by HarryWangATX — CHANGES_REQUESTED** [review]
add `## Implementation`

**Comment by maggie-j-liu** [comment]
@cskitty i still see this?
image

**Pull request merged**